npm/controlplane.go

```diff
@@ -0,0 +1,145 @@
+package npm
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	npmconfig "github.com/Azure/azure-container-networking/npm/config"
+	controllersv2 "github.com/Azure/azure-container-networking/npm/pkg/controlplane/controllers/v2"
+	"github.com/Azure/azure-container-networking/npm/pkg/dataplane"
+	"github.com/Azure/azure-container-networking/npm/pkg/transport"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/version"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+)
+
+type NetworkPolicyControlplane struct {
+	config npmconfig.Config
+
+	// tm is the transport layer (gRPC) manager/server
+	tm *transport.Manager
+
+	// Informers are the Kubernetes Informer
+	// https://pkg.go.dev/k8s.io/client-go/informers
+	Informers
+
+	// Controllers for handling Kubernetes resource watcher events
+	K8SControllersV2
+
+	// Azure-specific variables
+	AzureConfig
+}
+
+func NewNetworkPolicyControlplane(
+	ctx context.Context,
+	config npmconfig.Config,
+	informerFactory informers.SharedInformerFactory,
+	dp dataplane.GenericDataplane,
+	npmVersion string,
+	k8sServerVersion *version.Info,
+) *NetworkPolicyControlplane {
+	klog.Infof("API server version: %+v AI metadata %+v", k8sServerVersion, aiMetadata)
+
+	mgr := transport.NewManager(ctx, config.Transport.Port)
+
+	// TODO (nitishm): This needs to be passed to the Dataplane instance
```

Comment:
@vakalapa Here the input channel needs to be passed to the Generic(?)Dataplane instance since `ApplyDatapath` would write to this channel.

Reply:
I am working on Control Plane DP Shim and can push a skeleton along with GS Processor.